plugins/devtools-react/src/devToolsRouteDataRef.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createExtensionDataRef } from '@backstage/frontend-plugin-api';
+import { JSX } from 'react';
+
+/**
+ * Represents a DevTools route that can be contributed by extensions
+ * @public
+ */
+export interface DevToolsRouteData {
+  path: string;
+  title: string;
+  children: JSX.Element;
```

Comment:
Hmm, I think we should document this more clearly actually, but looking at this it'd be much better to rely on existing `coreExtensionData`. That'll allow things like extension middleware to wrap the React element, but more importantly the `coreExtensionData.routePath` is required to make it possible to route to sub pages.

Essentially, if a piece of data already exists in `coreExtensionData`, **never** make it part of custom extension data unless it's in a substantially altered form.